web/src/features/fireWeather/components/graphs/PrecipGraph.tsx

```diff
@@ -0,0 +1,364 @@
+      observedPrecips: _observedPrecips,
+      forecastPrecips: _forecastPrecips
+    }
+  }, [utcOffset, observedValues, forecastValues])
+
+  // Effect hook for displaying graphics
+  useEffect(() => {
+    if (svgRef.current) {
+      /* Clear previous graphics before rendering new ones */
+      d3.select(svgRef.current)
+        .selectAll('*')
+        .remove()
+
+      const svg = d3
+        .select(svgRef.current)
+        .attr('viewBox', `0 0 ${svgWidth} ${svgHeight}`)
+
+      const chart = svg
+        .append('g')
+        .attr('class', 'chart')
+        .attr('transform', `translate(${margin.left}, ${margin.top})`)
+
+      const context = svg // svg group for Y axis and its labels
+        .append('g')
+        .attr('class', 'context')
+        .attr('transform', `translate(${margin.left}, ${margin.top})`)
+
+      const legendMarginTop = chartHeight + 40
+      const legend = svg
+        .append('g')
+        .attr('class', 'legend')
+        .attr('transform', `translate(${margin.left}, ${legendMarginTop})`)
+
+      /* Create scales for x and y axes */
+      const xScale = d3
+        .scaleTime()
+        .domain(xDomain)
+        .range([0, chartWidth])
+      const xScaleCopy = xScale.copy()
+      const yScale = d3
+        .scaleLinear()
+        .domain([0, 100])
+        .range([chartHeight, 0])
+
+      observedPrecips.forEach(precip =>
+        d3Utils.drawVerticalLine({
+          svg: chart,
+          className: 'precipLine__observed',
+          xScale: xScaleCopy,
+          x: xScale(precip.date) - 2,
+          y1: yScale(precip.value),
+          y2: yScale(0),
+          testId: 'observed-precip-line'
+        })
+      )
+
+      forecastPrecips.forEach(precip =>
+        d3Utils.drawVerticalLine({
+          svg: chart,
+          className: 'precipLine__forecast',
+          xScale: xScaleCopy,
+          x: xScale(precip.date) + 2,
+          y1: yScale(precip.value),
+          y2: yScale(0),
+          testId: 'forecast-precip-line'
+        })
+      )
+
+      /* Render the X & Y axis and labels */
+      chart
+        .append('g')
+        .attr('class', 'xAxis')
+        .attr('transform', `translate(0, ${chartHeight})`)
+        .call(
+          d3
+            .axisBottom(xScale)
+            .tickFormat(d3Utils.formatDateInMonthAndDay)
+            .tickValues(xTickValues)
+        )
+      context
+        .append('g')
+        .attr('class', 'yAxis')
+        .call(d3.axisLeft(yScale).ticks(5))
+      context // Temperature label
+        .append('text')
+        .attr('y', 0 - margin.left)
+        .attr('x', 0 - chartHeight / 2)
+        .attr('dy', '1.3em')
+        .attr('dx', '0')
+        .attr('class', 'yAxisLabel')
+        .text('Precipitation (mm/cm)')
+        .attr('transform', 'rotate(-90)')
+
+      /* Render legends */
+      // TODO: We're going to have to look at using layouts moving forward to achieve the placement of objects. https://www.d3indepth.com/layouts/
```

Comment:
Oh man - here we go again! We really need to prioritise figuring out the layout.